ibis/expr/types/relations.py

```diff
@@ -210,33 +210,39 @@ def _ensure_expr(self, expr):
         else:
             return expr
 
+    @util.deprecated(
+        version="5.0", instead="use a list comprehension and attribute/getitem syntax"
```

Comment:
Shouldn't this be `4.1`? I'm usually a bit confused the version argument here.

Perhaps we should have two arguments, one highlighting the version we deprecated the API in `asof` and one for marking the version it's going to be removed at?

`util.deprecated(asof="4.1", removed_in="5.0", instead=...)`

Reply:
Good idea! I'll add those arguments.

Reply:
For clarity, I've been treating `version` like your `removed_in`.

Reply:
We can use `removed_in` to raise if we forget to remove the API before the release.

Reply:
How so?